### DISCUSSION TITLE
how to insert edit and delete button on aggrid

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#510] [Q&A] [answered]
can insert button between rows like this.  and how
```py
alldata = ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': ui.button("edit delete")},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
    ],
    'rowSelection': 'multiple',
}).classes('max-h-40')
```
and is there a table in nicegui like datatable to search and sort data?
thanks

**Comment by falkoschindler** [comment, marked as answer]
At the moment you can't add buttons in an AG Grid.
It might be possible to implement this feature in a later version of NiceGUI. But we're currently working on a new table element based on Quasar's [QTable](https://quasar.dev/vue-components/table) that will probably allow inserting all sorts of UI elements much more easily (#500).
